controller/engine_image_controller.go

```diff
@@ -475,6 +515,22 @@ func (ic *EngineImageController) getVolumesForEngineImageUpgrading(volumes map[s
 		if v.Spec.EngineImage != v.Status.CurrentImage {
 			continue
 		}
+
+		volumeAndReplicaNodes := []string{}
+		if v.Status.CurrentNodeID != "" {
+			volumeAndReplicaNodes = append(volumeAndReplicaNodes, v.Status.CurrentNodeID)
+		}
+		replicas, err := ic.ds.ListVolumeReplicas(v.Name)
+		if err != nil {
+			return nil, errors.Wrapf(err, "cannot get replicas for volume %v", v.Name)
+		}
+		for _, r := range replicas {
+			volumeAndReplicaNodes = append(volumeAndReplicaNodes, r.Spec.NodeID)
+		}
+		if isReady, _ := ic.ds.CheckEngineImageReadiness(newEngineImageResource.Spec.Image, volumeAndReplicaNodes); !isReady {
```

Comment:
You can use the same proposed wrapper function here, `CheckEngineImageReadinessForVolume(volume, node string)`
Volume is a string and refers to the volumeName.
Makes sense to move the wrapper function into the ds then.

Reply:
Thank you. Fixed